src/main/java/org/konstructs/grass/GrassActor.java

```diff
@@ -109,13 +107,13 @@ public void onEventBlockUpdated(EventBlockUpdated blockEvent) {
     public void onBoxQueryResult(BoxQueryResult result) {
 
         // Expect a 3x4x3 box
-        if (result.result().data().size() != 3*4*3) {
-            System.out.println("onBoxQueryResult: Error: Got a box with size " + result.result().data().size());
+        if (result.getBlocks().length != 3*4*3) {
```

Comment:
@nsg It think this shouldn't happen? If it does, I would consider it a bug in the server rather than something that you should need to check for here. Or are you asking for boxes of different sizes?